### PR TITLE
Simplify CI

### PR DESCRIPTION
**Pull request by shaobo-he-aws**
*Issue #, if available:*

*Description of changes:*

* Download Z3 directly instead of using the one in downloaded Dafny binary
* Do not download Z3 for DRT
* Try to use the build script instead of setting environment variables manually

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by aaronjeline — APPROVED** [review]
Any big reasons for this change beyond simplicity?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -19,21 +19,18 @@ jobs:
     steps:
       - name: Checkout cedar-spec
         uses: actions/checkout@v3
-      - name: Download Dafny
+      - name: Install Z3
         shell: bash
-        run: wget https://github.com/dafny-lang/dafny/releases/download/v4.0.0/dafny-4.0.0-x64-ubuntu-20.04.zip
```
Comment (cdisselkoen) [inline .github/workflows/ci.yml:24]: just curious how we get Dafny now (and which version) since I don't see a direct replacement for this line?

**Review by cdisselkoen — APPROVED** [review]

**Comment by shaobo-he-aws** [comment]
> Any big reasons for this change beyond simplicity?

No, there shouldn't be.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (shaobo-he-aws) [reply .github/workflows/ci.yml:24]: Dafny is installed via dotnet tool. I think GitHub Action Ubuntu images have dotnet preinstalled.

**Pull request merged**